sdk/app_cpu1/common/drv/eddy_current_sensor.c

```diff
@@ -9,36 +9,101 @@ void eddy_current_sensor_init(void)
 {
     printf("EDDY CURRENT SENSOR:\tInitializing...\n");
 
-    // Set sampling rate to 20kHz
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_1_BASE_ADDR, 20000);
+    // Default PWM carrier frequency for AMDC is 100 kHz
+    //   Triggering on both PWM high and low would be an effective sample frequency of 200 kHz,
+    //   either high OR low, being effectively 100 kHz sampling.
+    //   To meet this timing, SCLK would need to be VERY fast.
+    //   Therefore, set eddy current sensors to only sample on PWM high by default
+    //   with an SCLK frequency of only 2 MHz. Because this won't fit into the 10us interval,
+    //   every other PWM trigger will be ignored for an effective sampling rate of 50 kHz
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_1_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_1_BASE_ADDR, 2000, 270);
```

Comment:
Can you make these constants a `define` in this module's header file?

i.e., in `eddy_current_sensor.h`

```C
#define EDDY_CURRENT_SENSOR_DEFAULT_SCLK_FREQ_KHZ (2000)
#define EDDY_CURRENT_SENSOR_DEFAULT_PROP_DELAY_NS (270)
```

Probably have to then write these function calls on multiple lines since the constants are so long... :)

Reply:
This way, a user could also just call:

```C
eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_1_BASE_ADDR,
    EDDY_CURRENT_SENSOR_DEFAULT_SCLK_FREQ_KHZ,
    EDDY_CURRENT_SENSOR_DEFAULT_PROP_DELAY_NS);
```